### ISSUE TITLE
linkerd tap broken on gke with edge-19.8.1

### ISSUE THREAD

**Issue by siggy** [linkerd/linkerd2#3191]
## Summary

PR #3167 introduced a Tap APIService, and moved `linkerd tap` to use that new service. Linkerd `edge-19.8.1` includes this change. When installed on a default GKE `v1.12.8-gke.10`, it fails a `SubjectAccessReview`.

## Repro

On GKE `v1.12.8-gke.10`

```bash
$ linkerd install --controller-log-level debug | kubectl apply -f -

$ linkerd tap -n linkerd deploy/linkerd-controller
Error: Unexpected API response: 500 Internal Server Error ({"error":"SubjectAccessReview failed with: not authorized to access deployments.tap.linkerd.io: no RBAC policy matched"})
```

From the linkerd-tap controller logs:
```
$ kubectl -n linkerd logs -f deploy/linkerd-tap tap
time="2019-08-02T21:25:36Z" level=info msg="running version edge-19.8.1"
time="2019-08-02T21:25:36Z" level=info msg="waiting for caches to sync"
time="2019-08-02T21:25:37Z" level=info msg="caches synced"
time="2019-08-02T21:25:37Z" level=info msg="starting gRPC server on :8088"
time="2019-08-02T21:25:37Z" level=info msg="starting admin server on :9998"
time="2019-08-02T21:25:37Z" level=info msg="starting APIServer on :8089"
...
time="2019-08-02T21:27:08Z" level=debug msg="SubjectAccessReview: namespace: linkerd, resource: deployments, name: linkerd-controller, user: siggy@buoyant.io, group: [system:authenticated]" addr=":8089" component=apiserver
time="2019-08-02T21:27:08Z" level=error msg="SubjectAccessReview failed with: not authorized to access deployments.tap.linkerd.io: no RBAC policy matched" addr=":8089" component=apiserver
```

### Additional privileges

GCP appears to give additional privileges to my user when running kubectl commands, versus after I've authenticated as `siggy@buoyantio` inside the cluster:

```bash
$ kubectl auth can-i watch deployments.tap.linkerd.io --subresource=tap -n linkerd
Warning: the server doesn't have a resource type 'deployments' in group 'tap.linkerd.io'
yes

$ kubectl auth can-i watch deployments.tap.linkerd.io --subresource=tap -n linkerd --as siggy@buoyant.io
Warning: the server doesn't have a resource type 'deployments' in group 'tap.linkerd.io'
no - no RBAC policy matched
```

### Integration tests passing

Confirmed tap works on TravisCI integration tests because the Travis user has a ClusterRoleBinding giving it cluster-admin.

**Comment by siggy** [comment]
Confirmed tap works on TravisCI integration tests because the Travis user has a ClusterRoleBinding giving it cluster-admin.

**Comment by grampelberg** [comment]
To get this working, you can run:

```bash
kubectl create clusterrolebinding \
                $(whoami)-cluster-admin \
                --clusterrole=cluster-admin \
                --user=$(gcloud config get-value account)
```

There's some weird disconnect between SARs and the webhook auth configured in GKE clusters by default.

**Comment by siggy** [comment]
This is resolved via docs (linkerd/website#450) and better error messaging (#3196).
